### PR TITLE
[NEW] Drupal oAuth Integration for Rocketchat

### PR DESCRIPTION
**Pull request by Lawri-van-Buel**
@RocketChat/core 

Updates issue #1921

This adds an easy to use Drupal oauth integration path and is the rocketchat part of the new rocketchat drupal integration module on drupal.org (version 7.x-1.1) that will soon be released. It will allow you to use Drupal as a Login provider for your rocketchat users.

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
@Lawri-van-Buel thanks for contributing!  Do you happen to have a link to the Rocket.Chat Drupal integration module you mentioned?  I'd love to check it out.

File: .meteor/versions [hunk 1]
```diff
@@ -137,6 +137,7 @@ rocketchat:crowd@1.0.0
 rocketchat:custom-oauth@1.0.0
 rocketchat:custom-sounds@1.0.0
 rocketchat:dolphin@0.0.2
+rocketchat:drupal@0.0.1
```
Comment (geekgonecrazy) [inline .meteor/versions]: Can you undo all other changes to this file but this line here?  Ideally this PR shouldn't be updating other packages, it should only be adding the code needed for drupal.  We'd handle the updating of packages seperately

File: .meteor/packages [hunk 2]
```diff
@@ -4,38 +4,38 @@
...
 fastclick@1.0.13
-http@1.2.11
+http@1.2.12
 jquery@1.11.10
 less@2.7.9
 logging@1.1.17
 meteor-base@1.0.4
 mobile-experience@1.0.4
-mongo@1.1.15
+mongo@1.1.16
```
Comment (geekgonecrazy) [inline .meteor/packages]: Can you undo changes to this file?  Like mentioned in the other comment, ideally we'd handle these types of updates in a seperate PR.  This should only about adding drupal.  Thx :+1:

File: .meteor/versions (hunk 1 above)
Comment (Lawri-van-Buel) [inline .meteor/versions]: I agree I though I had done just that, Will do with the new update as its base.

**Comment by Lawri-van-Buel** [comment]
@geekgonecrazy My drupal 7 module is on the Drupal.org Repo, I am writing some documentation before I will release the drupal module (should be sometime the coming week) as for the development version its on our opensource [gitlab](https://open.040lab.com/drupal/rocket_chat/tree/7.x-1.x)

**Comment by Lawri-van-Buel** [comment]
@geekgonecrazy can you confirm the issues are resolved? or have I missed something?

File: .meteor/packages [hunk 3]
```diff
@@ -170,3 +170,5 @@ yasaricli:slugify
 yasinuslu:blaze-meta
 deepwell:bootstrap-datepicker2
 rocketchat:google-natural-language
+rocketchat:drupal
+rocketchat:i18n
```
Comment (geekgonecrazy) [inline .meteor/packages]: Was it necessary to add this?  I think our other packages might just add this as a dependency.


@RocketChat/core is there a reason it's not in the packages list?

**Comment by geekgonecrazy** [comment]
@Lawri-van-Buel awesome I'll check it out.  Just will want to get feedback from more of the team, then get it merged 👍

File: .meteor/packages (hunk 3 above)
Comment (Lawri-van-Buel) [inline .meteor/packages]: not sure. I just used the build example and added rocketchat:drupal using the `meteor add rocketchat:drupal` command. that I take it added the lines to the package. I am new to meteor, but expierenced in drupal so do not be afraid to tell me i did something wrong ;) I prefer to do it right than muddle on in a 'wrong' way.

**Comment by Lawri-van-Buel** [comment]
great I'll await yours and their review

**Comment by Lawri-van-Buel** [comment]
Is this now ready to be merged or do you want me to change something still?

**Comment by geekgonecrazy** [comment]
@sampaiodiego only change i'm not sure about is that rocketchat:i18n was added to .meteor/packages it wasn't before.

**Comment by sampaiodiego** [comment]
@geekgonecrazy only top level packages are added to `.meteor/packages` file.. since `rocketchat:i18n` is only added by `rocketchat:lib` it is not there.

but this doesn't explain why it was added after a `meteor add rocketchat:drupal`.. 

btw, I don't think by just having the file `packages/rocketchat-drupal/i18n/en.i18n.json` it should be used for translations.. I would recommend move this translation strings over `packages/rocketchat-i18n/i18n/en.i18n.json` file since we don't have packages specific translations at the moment :disappointed:

**Comment by Lawri-van-Buel** [comment]
Updated my fork with the removal of the i18n line in .meteor/packages and moving the translatable strings to the i18n module.

**Comment by engelgabriel** [comment]
@sampaiodiego please let's include this on the next candidate release.

**Comment by designgurudotorg** [comment]
This is amazing! Great work :)

**Pull request merged**